### ISSUE TITLE
Enhancement: Camera centered on player

### ISSUE THREAD

**Issue by PyroFlareX** [OpMonTeam/OpMon-Archive#101]
Always have the player character at the center of the screen if possible. Exceptions being when you deem it to be.

**Comment by milkyroute** [comment]
It has been the case some time ago. However, we changed for many reasons, one of them being that we had to create large maps with unaccessible places to hide black from the player. If you have some arguments to give, we will listen to them joyfully, but for now I don't think we will change the camera for now. Thank you anyway! I'm leaving this issue open for a while if you want to give some reasons or arguments.

**Comment by PyroFlareX** [comment]
I agree with your above reasons, but for the areas you want to hide, just make it so the camera doesn't move into those areas. That way, it is the best of both.

**Comment by milkyroute** [comment]
Isn't it already the case? The camera doesn't move into black areas, because it stops at the edge of the map. Except for the indoor maps, but it's normal, it's already the case in Pokemon. Or I misunderstood what you meant?
**Edit :** It isn't, after all. We will think about it, but I don't think it's a bad idea.

**Comment by PyroFlareX** [comment]
Like you said above, the camera doesn't center on the player. Instead the player has a weird space that can be described by testing. Maybe having the camera move with the player, but has collision detection at the edges, and tries to recenter on the player? That could work for implementation.

**Comment by milkyroute** [comment]
I will do a poll on the Discord server to know what people prefer, and then I will change it for the 0.17 if they want it.
Edit : The Discord server choose to change the camera.
